low-level/sensors/impl/centos_7/disk_monitor.py

```diff
@@ -246,15 +246,37 @@ def run(self):
             with self._drive_info_lock:
                 # Prepare drives to monitor
                 self._drives = self._get_drives()
-                # send msg for new drives
-                for drive_path in self._drives.keys():
-                    if drive_path not in self._existing_drive:
-                        drive = self._drives[drive_path][self.DRIVE_DBUS_INFO]
-                        resource_type = self._get_resource_type(drive_path)
-                        specific_info = self._get_specific_info(drive_path, self.DISK_INSERTED_ALERT_TYPE)
-                        resource_id = self._drive_by_path.get(drive_path, str(drive["Id"]))
-                        self._send_msg(self.DISK_INSERTED_ALERT_TYPE, resource_type, resource_id, specific_info)
-                        self._existing_drive.update({drive_path: False})
+                # Send msg for drives which were added or removed
+                # when sspl was inactive.
+                detected_drives = set(self._drives.keys())
+                existing_drives = set(self._existing_drive.keys())
+                removed_drives = existing_drives - detected_drives
+                added_drives = detected_drives - existing_drives
+
+                for drive_path in added_drives:
+                    drive = self._drives[drive_path][self.DRIVE_DBUS_INFO]
+                    resource_type = self._get_resource_type(drive_path)
+                    specific_info = self._get_specific_info(
+                        drive_path, self.DISK_INSERTED_ALERT_TYPE)
+                    resource_id = self._drive_by_path.get(drive_path,
+                                                          str(drive["Id"]))
+                    self._send_msg(self.DISK_INSERTED_ALERT_TYPE,

```

Comment:
by the way, this send_msg function as well doesn't have an impact and recommendation fields.
https://github.com/Seagate/cortx-monitor/pull/506/files#diff-b1d63e625b98aa2437d11fbaf42cd2767f6d50982926f394cae5fc1e2c25b2e5R930

Reply:
as you said for nw_cable @cdeshmukh, we will address this with other sensors.